claims.go

```diff
@@ -16,13 +17,13 @@ type Claims interface {
 // https://tools.ietf.org/html/rfc7519#section-4.1
 // See examples for how to use this with your own claim types
 type StandardClaims struct {
-	Audience  string `json:"aud,omitempty"`
-	ExpiresAt int64  `json:"exp,omitempty"`
-	Id        string `json:"jti,omitempty"`
-	IssuedAt  int64  `json:"iat,omitempty"`
-	Issuer    string `json:"iss,omitempty"`
-	NotBefore int64  `json:"nbf,omitempty"`
-	Subject   string `json:"sub,omitempty"`
+	Audience  string  `json:"aud,omitempty"`
+	ExpiresAt float64 `json:"exp,omitempty"`
```

Comment:
Thank you for submitting a PR.

I believe one of the goals here is to maintain backwards compatibility with the upstream repo. I.e., make it a drop-in replacement.

Instead of changing the `StandardClaims` struct, would it suffice to add a new struct with an explanation and/or a deprecation notice?

New users would be pushed towards the "correct struct", while existing users are unaffected but have the option to update their implementations?

Reply:
This would be one way to go. I'm feeling very urged to fix the current one but as we want to maintain the drop-in contract in order to ease the transition from the already deprecated and stale package we'll probably have to work around it this way until a new release is made in this one

Reply:
As you wish, but the current implementation is not valid according to the spec and fails hard in the wild (basically when interacting with PHP project, because Lcobucci's lib is the most popular in this ecosystem).

Reply:
I'd love to see this fixed as well. Although I think it's an easier migration story for users:

golang-jwt/jwt v1.x.x is backwards-compatible with dgrijalva/jwt-go (despite the incorrect implementation you pointed out)

and then we could consider further improvements for those already on /v1 that breaks compatibility with a /v2 release

Reply:
> I'd love to see this fixed as well. Although I think it's an easier migration story for users:
> 
> golang-jwt/jwt v1.x.x is backwards-compatible with dgrijalva/jwt-go (despite the incorrect implementation you pointed out)
> 
> and then we could consider further improvements for those already on /v1 that breaks compatibility with a /v2 release

I fear, that this won't be completely possible anyway, because the cleanest fix to #6 is to adjust the claims struct, changing `aud` from `string` to `[]string'. We might as well correct the float/int issue as well before releasing `v1.x.x`.

Reply:
Hehe, these are the trickiest tradeoffs. The breaking change is justified from a security and spec perspective, but keeping `StandardClaims` as-is maintains backwards-compatibility. The latter is something most Go folks have come to appreciate. Just want to make sure if we do break `StandardClaims` its worth it.

What are your thoughts w.r.t adding a new struct `RegisteredClaims` while keeping `StandardClaims` unchanged with a deprecation notice and comment _suggesting_ against its use?

Reply:
I will play around with this idea over the weekend. With that in mind I would then also probably choose to make the audience field directly a time Field instead of an into or float and have appropriate MarshalJSON functions that take care of the conversion. This feels more naturally and idiomatic to me than assigning the expiry in a float value.